packages/replay/src/auth.ts

```diff
@@ -263,3 +264,128 @@ export async function maybeAuthenticateUser(options: Options = {}) {
     return false;
   }
 }
+
+function parseId(id: string) {
+  return Buffer.from(id, "base64").toString("utf-8").split(":")[1];
+}
+
+async function getAuthInfo(key: string) {
+  const resp = await query(
+    "AuthInfo",
+    `
+        query AuthInfo {
+          viewer {
+            user {
+              id
+            }
+          }
+          auth {
+            workspaces {
+              edges {
+                node {
+                  id
+                }
+              }
+            }
+          }
+        }
+      `,
+    undefined,
+    key
+  );
+
+  if (resp.errors) {
+    throw {
+      id: "graphql-error",
+      message: resp.errors
+        .map((e: any) => e.message)
+        .filter(Boolean)
+        .join(", "),
+    };
+  }
+
+  const response = resp.data as {
+    viewer: {
+      user: {
+        id: string | null;
+      } | null;
+    };
+    auth: {
+      workspaces: {
+        edges: {
+          node: {
+            id: string;
+          };
+        }[];
+      };
+    };
+  };
+
+  const encodedUserId = response?.viewer?.user?.id;
+  const encodedWorkspaceId = response?.auth?.workspaces?.edges?.[0]?.node?.id;
+
+  return {
+    userId: encodedUserId ? parseId(encodedUserId) : null,
+    workspaceId: encodedWorkspaceId ? parseId(encodedWorkspaceId) : null,
+  };
+}
+
+function getAuthInfoCachePath(options: Options = {}) {
+  const directory = getDirectory(options);
+  return path.resolve(path.join(directory, "profile", "authInfo.json"));
+}
+
+function authInfoCacheKey(key: string) {
+  return createHash("sha256").update(key).digest("hex");
+}
+
+async function writeAuthInfoCache(
+  key: string,
+  authInfo: { userId: string | null; workspaceId: string | null },
+  options: Options = {}
+) {
+  const cachePath = getAuthInfoCachePath(options);
+  await mkdir(path.dirname(getAuthInfoCachePath(options)), { recursive: true });
+  const cache = {
+    [authInfoCacheKey(key)]: authInfo,
+  };
+  await writeFile(cachePath, JSON.stringify(cache, undefined, 2), { encoding: "utf-8" });
+}
+
+async function readAuthInfoCache(key: string, options: Options = {}) {
+  try {
+    const cachePath = getAuthInfoCachePath(options);
+    const cacheJson = await readFile(cachePath, { encoding: "utf-8" });
+    const cache = JSON.parse(cacheJson);
+    return cache[authInfoCacheKey(key)];
+  } catch (e) {
+    debug("Failed to read auth info cache: %o", e);
+    return;
+  }
+}
+
+export async function initLDContextFromKey(options: Options = {}) {
+  const apiKey = options.apiKey ?? (await readToken(options));
```

Comment:
It's also possible that the identity is in an environment variable so we should probably standardize on how we read this value to ensure we always end up with the same value.

Reply:
Created getApiKey function that reads --apiKey, env vars, saved token in this order.